zipline/data/data_portal.py

```diff
@@ -527,20 +538,26 @@ def _get_history_daily_window(self, assets, end_dt, bar_count,
                                 index=days_for_window,
                                 columns=None)
 
-        data = []
+        future_data = []
+        eq_assets = []
 
         for asset in assets:
             if isinstance(asset, Future):
-                data.append(self._get_history_daily_window_future(
+                future_data.append(self._get_history_daily_window_future(
                     asset, days_for_window, end_dt, field_to_use
                 ))
             else:
-                data.append(self._get_history_daily_window_equity(
-                    asset, days_for_window, end_dt, field_to_use
-                ))
-
+                eq_assets.append(asset)
+        eq_data = self._get_history_daily_window_equities(
+            eq_assets, days_for_window, end_dt, field_to_use
+        )
+        if future_data:
```

Comment:
We concat the data, so both would be returned; though this behavior is not covered by testing yet. (Figured out this out when the implementation was not attempting to concatenate at all.)